app/src/main/java/ch/epfl/sweng/SDP/utils/ImageSharer.java

```diff
@@ -0,0 +1,115 @@
+package ch.epfl.sweng.SDP.utils;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.facebook.share.model.ShareLinkContent;
+import com.facebook.share.model.SharePhoto;
+import com.facebook.share.model.SharePhotoContent;
+import com.facebook.share.widget.ShareDialog;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import ch.epfl.sweng.SDP.Activity;
+import ch.epfl.sweng.SDP.auth.Account;
+import ch.epfl.sweng.SDP.firebase.FbStorage;
+
+public class ImageSharer {
+
+    private static ImageSharer instance = null;
+
+    private Activity activity;
+    Context context;
+
+    ImageSharer(Context context, Activity activity) {
+        this.context = context;
+        this.activity = activity;
+    }
+
+    /**
+     * Get this ImageSharer instance.
+     *
+     * @param context context calling this method
+     * @return ImageSharer instance
+     */
+    public static ImageSharer getInstance(Context context, Activity activity) {
+        if (instance == null) {
+              instance = new ImageSharer(context,activity);
+        }
+        return instance;
+    }
+
+    /**
+     * Get the ImageSharer instance.
+     */
+    public static ImageSharer getInstance() {
```

Comment:
Why do you need this method? And if you really need it, why it has a different body from the other?

Reply:
This image is used to retrieve the instance if it's already created. I use this since you don't need to call using the arguments every time.